### ISSUE TITLE
Mail: New mail UI/UX for Usermin/Webmin

### ISSUE THREAD

**Issue by iliaross** [webmin/authentic-theme#431]
Hi,

Everyone is interested in new mail outlook please post your ideas. The main concept provided by me will be based on Google Mail. In case you have any ideas or proposals before I start bringing it to life, go ahead and post your ideas/proposals here.

This is the example of how I would like the new mail interface look like

![new_mail](https://cloud.githubusercontent.com/assets/4426533/14405661/d5fcd4b4-fea4-11e5-8a3d-aa7831dbd79b.png)

-----

**Mail related issues:**
  1. [Mailbox not refreshing using Authentic Theme:](https://www.virtualmin.com/node/55547)
  2. [Send mail doesn't work (timeout) when used in Safari](https://github.com/iliaross/authentic-theme/issues/1102)

**Comment by SteffanCline** [comment]
This is what you're making for Usermin? Looks nice. I've been using Roundcube for some time. There's a lot of good plugins for it with filtering etc. Saw an Outlook theme for it but $100 seemed steep. This is definitely nice. Might I suggest using TinyMCE or CKEditor (http://ckeditor.com/) for the HTML?

**Comment by singhdd93** [comment]
Really Love the new design ...
I think CKEditor is much better than TinyMCE.

**Comment by qooob** [comment]
Yes, I will use `CKEditor` too and most likely will use it.

**Comment by GMo888** [comment]
The interface seems to me very well.

It would be nice to have the opportunity to right clicks on the subject of mail to reply, forward ... A bit like Outlook Web Access ...

**Comment by qooob** [comment]
Sure, context menu will be there for sure!

**Comment by divyeshsachan** [comment]
Interface is looking decent in attached pic.
Here are few suggestions:: Provide drag & drop for easy files attachment, multiple mail select + drag & drop to folder for easy mail movement across folders.
As per my opinion instead of three column view for writing mail, mail writing should be in dialog (as like in file editor in your authentic file manager) or as dialog in like gmail in bottom right corner or in full second column. The point is, while drafting a mail, one rarely look at his list of mails in any folder and if it's in three column view second column of mail list will unnecessarily occupy much screen space.

:-)

**Comment by qooob** [comment]
I will consider all of your proposals. I will however and most likely make that _gmail like_ composer optional..I like the way it is on the screen shot.

The theme will just inject additional options to the module's configuration.

**Comment by apurva577** [comment]
The Mailbox looking very rich and user friendly. I hope it can have the ajax functionality like gmail and roundcube. It will add an biggest punch on it.

By when it will have its launch or first release ?

**Comment by qooob** [comment]
I'm starting working on new graphs now and perform other ongoing improvements.

While doing it, I'm gonna submit a request to Webmins/Usermin back-end support for getting data in JSON format. Whenever it's ready, I'll start.

Yes, it'll be ajaxed surely. I'll try making it as good as gmail.

**Comment by aminkh17** [comment]
+1

**Comment by qooob** [comment]
Work in progress. My plan is to release it by the end of summer.

**Comment by Stinkykiller** [comment]
It would be nice if the theme in Usermin can pull email + aliases of specific virtualserver and show this in "From:" like drop down menu. Something like this:
- main VS email
--- alias 1
--- alias 2
--- alias 3
etc.
or in case of 2 or more emails for specific VS:
- VS email A
--- alias a1
--- alias a2
--- alias a3
- VS email B
--- alias b1
--- alias b2
--- alias b3
etc.
Leave the tab "From" in case someone want to insert custom email but on main window where the email is composed it would be nice to have this option.

**Comment by qooob** [comment]
When I start working on the email, I will take all considerations and try to make it best.

Thanks.

**Comment by Stinkykiller** [comment]
Maybe i should post this suggestion on Vmin site, not sure if this should go as Usermin or Authentic feature. Your opinion?

**Comment by qooob** [comment]
Hard to tell. Most likely, I will make mail work as an extension (like File Manager) as it gets very hard to collaborate thoroughly on this kind of matters.

Ideally it would be best to separate logic and presentation _completely_ but it's not going to work fully for our project.

Later my extensions could be merge with Webmin branch/modules no problem.

Doing this way provides much faster access to new features and let's me faster enroll new features.

**Comment by qooob** [comment]
Estimated time for newish Mail support is 18.40+

**Comment by Stinkykiller** [comment]
Forgot to mention in my earlier messages. Design looks great but it would be nice to offer few color schemes because not everyone is fan of bright colors. Aside of that i like it.

**Comment by qooob** [comment]
Alright, I will consider it. ;)

**Comment by apurva577** [comment]
Hi Bro... First of all I Really appreciate your work and I inspired alot from you.

I have few more suggestions for new mail interface.

1) Generally to access mail user needs to login webmin. It would be great to have the functionality to set the subdomain from root user to access new mail interface like webmail.example.com or in domain base url like example.com/webmail which displays login screen to access only mails like roundcube.

2) Another option to backup all mails and can import that backup on another server having virtualmin/cpanel which helps to easily migrate in case of host transfer.

By when its expected the first version of new mailing module to be get released ?

**Comment by qooob** [comment]
Hi,

I'm actually about to start, pretty much right after Webmin 1.830.

About your points, I must say that they are all already there. For the first, Usermin exists for user-level access to the server. For the second, backups can be easily done by copying _Maildir_ folder to the other server.

**Comment by bobrocksgithub** [comment]
Hi qoob, I do have one request (ok 2):
1) Enable user-selectable colors for the login page (and keep the ability to use a custom logo)
2) Enable the ability to deactivate the 'usermin' logo and text, so that only the custom logo is visible at login...

Your template for mail looks good to me!

**Comment by qooob** [comment]
I'll consider it. Thanks.

**Comment by bobrocksgithub** [comment]
If it helps, the login for nextcloud looks great. 
They keep the branding on the bottom, but allow the page to be fully customized.
Thanks for your excellent work and your consideration!

**Comment by qooob** [comment]
Hi,

Sure, I will take a look.

**Comment by topstarnetwork** [comment]
I can not waite to see it on production server, I love it!

**Comment by tpnsolutions** [comment]
UI concept is looking pretty sexy! Ofcourse, coming from you bro it's no surprise :-)

**Comment by qooob** [comment]
It will be even hotter. :D It's just a prototype.

**Comment by apurva577** [comment]
When the mail module gets release ?

**Comment by qooob** [comment]
@apurva577 The work is in progress. I have started working on HTML part. As I integrate it into the theme, then I will do scripting for both server and client. It's a long time work.

I can't do any estimations. I tend to do things perfectly without thinking of resource, meaning time-wise or other.

I might post additional video or screenshots here to see the reaction.

**Comment by forofo** [comment]
very nice design. Just I think it would be good having more messages in the screen; perhaps an smaller font, or at least some option to change the size. 

congratulations!

**Comment by qooob** [comment]
@forofo Sure, all of it will be there. The reason why theme is what it is today, is because I'm actually using it myself. Same about the mail. I'm going to use new Mail Interface for personal use. Find bugs and ways for constant improvements.

**Comment by qooob** [comment]
Question to all who is reading it.

Do we keep in places **_the left menu_** in Usermin for Mail and only re-working the right side or we change it completely making it look like a separate app, something more like on the screenshot? By the way, it's going to be possible to add all folders to the current left menu while keeping overall the same design of Webmin/Usermin?

**Comment by divyeshsachan** [comment]
Both will be good, depends on user interest. From my perspective, everything should be on right side and keeping left side as it is or you can also a add hide option for left menu while in mail.
Lets see what other thinks. 
Really appreciate your efforts.

**Comment by jcameron** [comment]
Personally I don't like those three-column designs, as they look cramped on smaller monitors / browser windows.

**Comment by qooob** [comment]
OK, it was just an example. What about having `New email` as new window (that could be maximized), like in GMail? Improving right side and keeping left for regular stuff, like links and folder names (including mulitple level)?

**Comment by swelljoe** [comment]
All the elements look great, but I have concerns with the width of it. When I look at the full-size screenshot 80% of the message window is off-screen (I have my screen split roughly in half with the browser in half and my editor and terminal in the other half). So, two panes is better, IMHO, as with GMail and Google Inbox.

I have been thinking we may want to spin off the webmail into its own thing...completely leave Usermin behind, and use the JMAP proxy I talked to you about a while back  as the backend. We could still serve it from Usermin, I guess, but really any web server would do, since the frontend would do all of the work and it would authenticate to the JMAP backend.

If we're rebuilding everything from the ground up, might as well strip down to a core thing that can be used as widely as possible. Depending on Usermin isn't necessary or useful if the frontend is doing a bunch of work and we want the frontend to work with JSON objects from the mail server.

**Comment by swelljoe** [comment]
And, as JMAP finds its way into IMAP servers (Dovecot is still in dev, Cyrus already has beta support, and we could ship Cyrus instead of Dovecot, if necessary...both are very good) we'll already be ready for it.

**Comment by jcameron** [comment]
I like the gmail approach for new mail - a sub-window that pops up over part of the mail list. Or better still, do it the way gmail used to and cover the entire mail list when composing email.

**Comment by qooob** [comment]
I will stick with GMail style and try to make things optional to satisfy different people's predilection.

**Comment by gnadelwartz** [comment]
i like the idea to seperate it from usermin as a standalone webmail, but also with full intrgration if usermin is installed. it may then replace my roundcube installation 😁

anything new?

**Comment by qooob** [comment]
Yes! I'm finishing bug fixes Servers Index and [with downloads](https://rostovtsev.ru/pub/src/authentic/19.05/multiple_downloads_progress_example.webm), adding multiple uploads to File Manager and other modules **and starting on the Mail**. I can't wait to start working on Mail. It looks so ugly in Authentic Theme.

**Comment by necrophcodr** [comment]
@qooob as a client of virtualmin, i hope that the new Mail system will continue to preserve the exact functionality at least, since many features are used by our clients, not limited to just auto reply and the vast filtering options with Procmail.

**Comment by qooob** [comment]
We surely going to expand functionality, not cut it down.

**Comment by JimTR** [comment]
my input :- let it have the same functions as rainloop .. it would then be awesome

**Comment by swelljoe** [comment]
@JimTR can you be more specific? I don't know of any features in Rainloop that aren't already in Usermin (but I don't use Rainloop, have only looked at it in passing). Rainloop looks nice, of course, but we'll need to know what functions you have in mind before we can think about implementing them.

**Comment by JimTR** [comment]
Rainloop allows you add different domains to the client E.G gmail, hotmail etc, it then gives you a dropdown to switch accounts ... I have about 8 domains set up some on the local (to rainloop) server some on another postfix server and gmail etc .
Add a user Identity function to make the the 'from email address' to match from one of your other email address to save switching between accounts to send an email

**Comment by JimTR** [comment]
After playing with both Rain Loop and usermin for a couple of hours ... Usermin does most everything else and perhaps a bit more ... E.G the ability to change password ... however the ability to add another email server with rain loop is admin only (I agree with that) however a user can add an existing email account from a ratified email server

**Comment by necrophcodr** [comment]
@JimTR I don't think a usermin skin should handle more domains. Why would that make sense, if it's made for handling the local mailbox system? It makes more sense to setup Rain Loop if you wanted to manage the emails of multiple internal and external domains, as usermin manages only one.
Not that I have a say, I'm just commenting on the logic of this.

**Comment by qooob** [comment]
We will do our best to implement all possible required functionality. Not everything should be done on the theme side.

When we get to the stage, where everything works as it is now, we'll start adding functionality. Multiple identities sounds like must have feature.

**Comment by swelljoe** [comment]
Multiple accounts would definitely need to happen in Usermin itself rather than theme), but there's nothing wrong with discussing it here. I'm following the conversation and can help with the backend stuff that needs to happen to make good ideas happen (and multiple accounts is a good idea).

**Comment by JimTR** [comment]
@necrophcodr Personally I would perfer to usermin over Rain Loop as of it's tight intergration to webmin but I do use the multiple domain option of rain loop as I have all my mail in one place this married to the identity functions allows me to send mail from any of my email addresses in one hit. 
I thought that Usermin was a mail client and not just a webmin module

**Comment by JimTR** [comment]
@qooob have a go with Rain Loop as style wise they appear to have the appearance nearly right ... I tend to use the blur theme over the standard one

**Comment by qooob** [comment]
@JimTR I'll make the mail shine. Usermin going to have most awesome mail UI. Just slightly more time. To release 19.09 (bug fix) and 19.10 new feature release. Then Mail.

**Comment by necrophcodr** [comment]
My only concern with it was the users. If a user has 5 domains on their account, and they log in to one of the Usermin accounts for one of the users on one of the domains, they may add emails for other accounts on the other domains as well and access them from various places.

I only fear that it will confuse ordinary folk.

**Comment by qooob** [comment]
I think we do this on example of Google Mail.

You can add multiple aliases/identities to your existing account, as long as you could verify them.

**Comment by JimTR** [comment]
@necrophcodr 
I use Thunderbird as a desktop mail client and have exactly the same email accounts connected .. it is not confusing. Perhaps think of usermin as a mail client rather than an extension of webmin

**Comment by tpnsolutions** [comment]
How bout we just call the new app... "Mailman"...

**Comment by qooob** [comment]
I don't think so. It should be Usermin with new UI. There is no reason to call it differently.

**Comment by tpnsolutions** [comment]
@qoob,

If you go into Gmail settings, they have the ability to poll external email accounts. I'd have seconded the option of "sending as" an alternate address, though you'd likely have to build in some SPF and DKIM verification to make this useful as sending as say bob@gmail.com from your server would set off red flags as it would be seen as spoofing by many providers.

**Comment by tpnsolutions** [comment]
@qooob,

I was joking about the name change.

**Comment by qooob** [comment]
@tpnsolutions We surely will add identities. Fetchmail already can grab mail from other accounts.

**Comment by swelljoe** [comment]
Usermin has IMAP support already; the only thing missing is multiple accounts. It's actually probably not hard to implement multiple accounts, though I haven't looked at it, or really thought through how many pieces need to be changed.

I can foresee a unified inbox being tricky-ish to implement well. It'd require a lot of additional meta-data be passed around so that when you click "Reply" it knows to reply from the right address, among other things. It probably also needs to send via the right SMTP server in such a case, too, since some servers will be configured to refuse to send mail from domains that aren't locally hosted, and likewise, some servers will be configured to refuse mail sent through a server that isn't in the SPF and DKIM records.

**Comment by jcameron** [comment]
FYI, Usermin does support multiple IMAP accounts already - you can create additional folders that are backed by different accounts.

**Comment by rostovtsev** [comment]
> ..though you'd likely have to build in some SPF and DKIM verification.

I have done _SPF_ and _DKIM_ verification test, along with showing _signed by_ label, and it will be displayed as a lock/unlocked icon with different color based on the incoming message header. Going to be useful.

For those who is interested, I'm in the middle of making mail listing.

-----

Anyone reading it. 

Please vote below for _How mail preview should be done?_

**Comment by rostovtsev** [comment]
Please vote using thumb-up/thumb-down:

**No preview** - _like the way it is now in Usermin ad GMail by default, with just a table and showing mail only when clicking it_

**Comment by rostovtsev** [comment]
**Privew on the right* - _like Hotmail for example. Mail listing on the left and the preview next to it, on the right (like in the screenshot at the very top of the thread)_

**Comment by rostovtsev** [comment]
*Preview on the bottom* - _self explanatory_

**Comment by rostovtsev** [comment]
Please welcome new **mail** with working listing and controls. It's been _many weeks of work_, and only now I'm ready to release the first beta.

**What is working:**

 1. Mail listing;
 2. Mail sorting by date, sender, size, subject (possibly other, such as spam score);
 3. Selecting mail based on type; deleting, reporting as spam/ham; whitelisting/blacklisting;
 4. Moving/copying mail;
 5. Paginations with inversion of listing (like Google Mail);
 6. Marking mail as starred/unstarred (ex. special status), just by clicking the star icon;
 7. Gravatar is displayed in case available for the sender;
 8. Displaying basic mail headers, including a check for sender's signature and encryption status;
 9. Date is displayed using _from now_ format and complies with recently added locale feature.

**What is not working** _yet_:
 1. Error handling;
 2. Mail forwarding;
 3. Mail searching.

**Example of how it looks:**

![screenshot from 2018-07-25 19-45-50](https://user-images.githubusercontent.com/4426533/43215920-24d2ead0-9046-11e8-8646-b5b736a84666.png)
![screenshot from 2018-07-25 19-46-12](https://user-images.githubusercontent.com/4426533/43215921-24ef5814-9046-11e8-8869-6d3570e731f3.png)

-----

I encourage you to give it a try and share your experience. :heart:

**Comment by rostovtsev** [comment]
JavaScript [source file](https://github.com/authentic-theme/authentic-theme/blob/master/extensions/mail/mail.src.js) available as well.

**Comment by JimTR** [comment]
HI I have been using usermin new theme from release, it's ok but has a few bugs (should I open issues for each or carry on here ?) and I really do miss the preview offered with other clients, is there a chance a setting could be done to allow you to add previews in whichever place you want (right,bottom)

**Comment by rostovtsev** [comment]
Yes, previews are going to be implemented a bit later. You can post bugs here.

**Comment by JimTR** [comment]
Bug 1
As you can see from the immage everyone (nearly) has the same gravitar image
![umin1](https://user-images.githubusercontent.com/4038510/43675906-f672df32-97df-11e8-93ce-ea5c0f1b6537.png) which I don't think is correct

**Comment by JimTR** [comment]
I know that to be my gravitar .. I have tried incognito mode and its the same however I can not right click to inspect the element (to find the url) as the message just gets selected .. I will view the source of the page to see if I can find it

**Comment by rostovtsev** [comment]
You won't find anything in the page source. You'd need to use browser's console.

**Comment by JimTR** [comment]
Look through the console and there is nothing regarding this ... only a couple of errors regarding an image being rendered over http rather than https ... I'll start looking at headers to see if there is anything common with the effected messages

**Comment by rostovtsev** [comment]
This is odd, as image served as `src="//www.gravatar.com/avatar/..."`. Could you try other PC or/and a tablet or a phone?

**Comment by JimTR** [comment]
I could boot the same pc with windows rather than ubuntu

**Comment by rostovtsev** [comment]
Okay. That would help. Thanks.

**Comment by JimTR** [comment]
Oddly enough I set up another email & gravitar image and that displays as it should .. my best guess is - if the from email address contains your email address the code is picking that.
E.G your email address is fred@domain.com and the from email address is bounce.fred@domain.com ... not sure at this point but I'll boot into windoze to see if there is any change

**Comment by rostovtsev** [comment]
No, that wouldn't be possible as emails a hashed.

**Comment by JimTR** [comment]
ok console from windows gives -
Uncaught TypeError: $(...).fancytree is not a function
    at Object.tree (<anonymous>:1:21107)
    at Object.set (<anonymous>:1:22865)
    at render (<anonymous>:1:20252)
    at Object.success (<anonymous>:1:7236)
    at c (bundle.min.js?1920153338383501:3)
    at Object.fireWith [as resolveWith] (bundle.min.js?1920153338383501:3)
    at r (bundle.min.js?1920153338383501:4)
    at XMLHttpRequest.<anonymous> (bundle.min.js?1920153338383501:4)

**Comment by rostovtsev** [comment]
Yes, it happens under certain circumstances for now. It's more or the less harmless, and meanwhile to work around, you need to click on any _folder name_ in navigation menu.

**Comment by JimTR** [comment]
This does not resolve the gravitar issue .. this was the only notification in the console

**Comment by rostovtsev** [comment]
It has nothing to do with gravatar.

Okay, thanks for your feedback. I will fix the console error on next the beta3.

**Comment by rostovtsev** [comment]
Authentic Theme _19.20-beta3_ brings new mail search; improved mail listing display and many other improvements.

![screenshot from 2018-08-23 20-22-52](https://user-images.githubusercontent.com/4426533/44542293-0b44a500-a715-11e8-90de-c688135a336a.png)


In order to install _19.20-beta3_ you would have to install latest Git version of _Webmin/Usermin_.

**Comment by JimTR** [comment]
looks good but I would suggest changing "has the words" to "contains" and "does not have" to "does not contain"

**Comment by rostovtsev** [comment]
Thanks Jim, for your idea. I will consider it.

**Comment by JimTR** [comment]
I've just got the gravtar thing ... it is based on the 'TO:' email address and not the 'FROM:'  so therefore you will always get your avatar and not the senders avatar .. ok I now understand

**Comment by JimTR** [comment]
now the console is fixed I get the following error type :
Mixed Content: The page at 'https://noideersoftware.co.uk:20000/mailbox/view_mail.cgi?start=0&id=18294&folder=0&xnavigation=1' was loaded over HTTPS, but requested an insecure image 'http://uacfmail.s3.amazonaws.com/images/UACF_MFP_logo_mfp.png'. This content should also be served over HTTPS.
perhaps this is related

**Comment by rostovtsev** [comment]
Okay, I'll double check avatar implementation.

About the error, it's coming from your side, by the way, so when you embed something to the theme, start URL with `//yoururl.com/..`, avoiding protocol, and thus it will be set correctly automatically.

**Comment by rostovtsev** [comment]
Where do you use this URL? Is it in mail view?

**Comment by JimTR** [comment]
rather than using https://yoururl.com:20000 I used //yoururl.com:20000 however the embeds you mention are loaded via usermin in mail content, this does not effect usermin in anyway but throws errors in the console .. I coded some thing in php to fix this for another project which you could recode into whichever language you use which may reduce console errors

**Comment by rostovtsev** [comment]
No, I mean for your `amazonaws` link.

**Comment by JimTR** [comment]
That is not mine that is in the content of a received message .... the message writer has added a url that is not HTTPS so HTTP ... but usermin retains secure mode (padlock) good 
![errormail](https://user-images.githubusercontent.com/4038510/45507188-4c1a6180-b789-11e8-9b78-6287980ad7b4.png)
sort of notes this

**Comment by JimTR** [comment]
looking at the same message with the console switched off
![samemessage](https://user-images.githubusercontent.com/4038510/45507924-4b82ca80-b78b-11e8-83fc-a0eb48f6670e.png)

**Comment by rostovtsev** [comment]
Like I said, it's going to be reworked (and fixed) before 19.20 final release.

**Comment by JimTR** [comment]
Ok 
I have been using usermin over the last week and it does need some extra functions/changes

1. The gravatar pointing to reciepent  address sort of works as a spam filter .. as the code reads the first email address in a list .. so if I don't see my avatar I know the mail was addressed to multiple people and is probably spam.

**Comment by rostovtsev** [comment]
> 1. The gravatar pointing to reciepent address sort of works as a spam filter .. as the code reads the first email address in a list .. so if I don't see my avatar I know the mail was addressed to multiple people and is probably spam.

This [has been fixed](https://github.com/authentic-theme/authentic-theme/commit/a97b4d93a2e80c05220e8a4798e6f150c3074f17). Thank you! :smiley_cat: 

Point _**2.**_ is in todo.

Issue _**3.**_ is a known bug and will be fixed in the near future.

**Comment by ksihota** [comment]
The Search threw me for a while as I kept getting an error when I searched for items:
![image](https://user-images.githubusercontent.com/29150649/50577222-970fc180-0dd8-11e9-8128-8e0f3a1826f6.png)
This showed when I was in the extended search and a 'No message matched your search' when using the single entry search. 
![image](https://user-images.githubusercontent.com/29150649/50577273-dc80be80-0dd9-11e9-868d-bb65d7f7a944.png)

I finally figured out that the 2 options related to the Subject Line only and not the body of the message. I think that this needs to be made clearer in some way.  The initial Subject Field line seems to be redundant here as there are now 2 other options (Contains and Does Not Contain) unless I am missing something.
I am also curious about the 2 different error messages that show depending on using the sigle field search or the fuller search option.

![image](https://user-images.githubusercontent.com/29150649/50577210-2e284980-0dd8-11e9-921c-bffb44779dff.png)

I also like to have a larger preview of the message body and was wondering if it is possible to add this using the same technique you have applied when you mouse over the sender to see the additional header information.  It allows the user a quick view of the content without having to wait for a page reload and keeps the screen from getting too cluttered.

Thanks for all your great work.

**Comment by ksihota** [comment]
Please excuse my ignorance here but this item caused me some concern until I figured out what I think was going on.
![spf error](https://user-images.githubusercontent.com/29150649/50616660-42cf1500-0e9e-11e9-81f6-f83f59707409.png)
Because the SPF showed received-spf in red with an x i figured my SPF was not working.  I finally figured out that the SPF check did not appear in the header of locally delivered mail (at least that's my guess) since I was sending and recieving using 2 different domains (virtual servers) that exist on the same server.
When I checked the headers I saw no mention of SPF and I guess that's why it displays the way it does. When I sent to an external account (my google email) I saw the proper SPF message showing the SPF Pass.
Is this something that needs to be considered on the theme or something that needs to be addressed in Usermin, or something that is just a waste of time?

**Comment by rostovtsev** [comment]
@ksihota Hi! 

> The search threw me for a while as I kept getting an error when I searched for items:

Yes, the error message is known issue and it should be addressed on Usermin side. I let @jcameron  know about it.

> I finally figured out that the 2 options related to the Subject Line only and not the body of the message.

Are you saying that _Has the words_ and _Does not have_ get only related to the Subject? If so it's the core bug and also should be addressed in Usermin. It should be addressed to the _whole_ message.

> I also like to have a larger preview of the message body

You will. I plan to continue working on Mail after holidays. I will try making it the best possible way. :smiley: 

> I finally figured out that the SPF check did not appear in the header of locally delivered mail

Yes, correct. SPF is not applied to the local mail. Maybe it shouldn't be displayed at all?

**Comment by rostovtsev** [comment]
> Most think that this search is for the messages. Wouldn't this search be more appropriate on the Usermin tab rather than the Mail tab? Or maybe it needs some kind of better heading? I never use it so I don't really see its purpose.

Actually, this search was meant to filter current messages' list, that is displayed on the right. It will be improved, and filtering of folders will be added to it as well.

> I also like the ability to see the DKIM and SPF in the dropdown. 

Of course it's not gonna go, it will not be displayed for local messages, or will show an appropriate message for locally delivered mail.

> From what I can tell the search never searches the body only the text in the Subject line. I checked the two other search fields again and it appears that they don't work for me.

There are issues in search, and we are working on it.

**Comment by ksihota** [comment]
I figured out what was causing the slow display while trying to locate the buttons on the Mailbox screen. 
Since all the buttons disappeared I hadn't noticed that the extra choices only showed up after selecting a message. Once I realized that, I found that the buttons were replaced with the menu.  Much cleaner design.
The actual slow performance I was having was due to me setting my mail to show 1000 messages at a time.  Obviously it was slow  loading that many (actually anything much over 100) so I set it back to 20 and it is working well.  The issue I have with loading only a few at a time is having to always page through the groups and more annoyingly is that when I choose a Select All it only selects the number that are currently displayed.  It would be nice to be able to have a **Select All Loaded** and **Select All** (which actually does select ALL)

Thanks for all your hard work

**Comment by rostovtsev** [comment]
It would be nice to be able to have a Select All Loaded and Select All (which actually does select ALL)


Good idea. I'll check how it could be done right after code refactor/optimization.

Thanks!

**Comment by ksihota** [comment]
Not sure if this is a theme issue or part of usermin but it is a slight detractor.
I use the Mark as Spam and Delete feature by first moving messages to a separate folder and then eventually Selecting All and marking and deleting the messages.  This allows me to use other mail clients on a regular basis and then complete the task in Usermin whenever I happen to use it.
What I see happening is that (now viewing only 20 at a time) I have to select groups of 20 to delete (which is working fine) but after I do so the list does not update in any way.  I can use the refresh icon on the right, by the message count, to refresh the view and see that something actually happened but I didn't realize this initially. I was sure it wasn't working when I had not seen any change in the message list or the message count. After playing around I realized it was just not refreshing the view and not providing any other visual feedback that the task had actually completed.
Not a big issue but maybe somthing that could be altered?

**Comment by rostovtsev** [comment]
@ksihota I really tried to understand what you meant but I'm still not sure that I did.

Do you want messages in the folder to be automatically updated?

**Comment by ksihota** [comment]
When I select all messages in the list and then choose Mark as Spam and Delete, nothing appears to happen except the messages uncheck. I would have thought that there would be some feedback ("Task Completed" etc) and then the deleted messages should disappear and the next group should appear if they exist.  If all messages are deleted the screen should be empty and the message count should reflect this.
That was not happening. 
If I select all again and complete the task a 2nd time (without refreshing) the same thing happens.  However, if I select the refresh button on the right, the messages disappear and the count changes. I tried doing the same set twice in case the selection was actually selecting messages that I didn't see on screen, but the 2nd attempt did nothing, only the initially selected messages were deleted. 40 selected in 2 steps, 20 deleted.
I believe that, really all that needs to be done is, a refresh needs to be included after the delete is processed so the user does not have to select the refresh manually.  A message could be added but it might be more of a nusance. There just needs to be some indication that the task was actually completed. 
The message count also shows this on refresh.
Hopefully this clarifies what I was trying to say.  Let me know if there is anything else that isn't very clear.
Great job on the changes.  They really make it easier to deal with the webmail.

**Comment by rostovtsev** [comment]
> When I select all messages in the list and then choose Mark as Spam and Delete, nothing appears to happen except the messages uncheck. I would have thought that there would be some feedback ("Task Completed" etc) and then the deleted messages should disappear .....

Yes, understood and confirmed - it's a bug. It's just been fixed on this commit: https://github.com/authentic-theme/authentic-theme/commit/28944b2149ba4ab4c80339c65bf509e24cfdd655.

> Let me know if there is anything else that isn't very clear.
Great job on the changes. They really make it easier to deal with the webmail.

Thank you! My pleasure! :smiley_cat: I love working on mail. I will regenerate my energy and continue with mail view and compose.

----

I am locking and closing this thread. You are welcome to open separate issues.

**Comment by rostovtsev** [comment]
I have been working a lot for the past few weeks to deliver enhanced UX, when working with mail. The new _mail composer_ is coming soon to version _19.40-beta1_.

**Examples:**
![image](https://user-images.githubusercontent.com/4426533/64477881-42171d00-d1a9-11e9-96ac-f026587b4d73.png)


![image](https://user-images.githubusercontent.com/4426533/64477718-902b2100-d1a7-11e9-84e4-9abba0de8d88.png)

![image](https://user-images.githubusercontent.com/4426533/64477911-89051280-d1a9-11e9-8e71-da763fc5366f.png)


![image](https://user-images.githubusercontent.com/4426533/64477824-ac7b8d80-d1a8-11e9-8daf-056a061d2994.png)


---

While it's in production, influence or share your ideas.

**Comment by ksihota** [comment]
Looks good from the screen shots. I like the idea of a scheduled send.
I would still like some way of selecting more than just the visible messages when you do a select All.
Is there some other way of doing that other than selecting 20 or so at a time or changing the preferences to display all messages on the page at one time?

**Comment by rostovtsev** [comment]
> Is there some other way of doing that other than selecting 20 or so at a time or changing the preferences to display all messages on the page at one time?

Yes, I think it's possible, just the same way I recently did for File Manager.

Do you need this for mass deletion? 

🙂

**Comment by ksihota** [comment]
Yes, basically. I have spam files placed in a folder and every so often I send them to mark as spam and delete. I also have a bulk folder, which is essentially junk mail that I might want to look at, and it is much easier to just clear the whole folder at one go.
I used to have all my messages visible, rather than just 20, but this was causing speed issues so I set it back to 20.

**Comment by rostovtsev** [comment]
The new _**Mail Composer**_ has been added as part of _19.40-beta1_ version. You are encouraged to try it and share your experience. It took many hours to code, though it's only [1000](https://github.com/authentic-theme/authentic-theme/blob/master/extensions/mail/mail.src.js#L798-L1557) lines of code. :sweat_smile: 

*Note:* In case to make server attachments work properly, you would need to update `mailbox` module from _Usermin_ repo as well, as there were few [bug fixes](https://github.com/webmin/usermin/commits/master) in _Usermin_ itself.

----

**What is working:**

  1. Mail composer, format and send;
     1.1. Insert images from clipboard;
     1.2. Link insert
     1.3. Inserting images inline from image upload
  2. Adding multiple attachments from user PC and from server (if enabled in settings);
  3. Email options, such as requesting read receipt, add recipient to address book and etc.;
  4. Setting message Priority;
  5. Encryption (if there is a valid GPG key present);
  6. Automatic save email as draft;
  7. Message progress while sending;
  8. Error handling and notifications.


**What is not working yet:** (but will in the very near future)

   1. Scheduled emails;
   2. On fly toggling HTML/plain-text modes;
   3. Message reply/forward still uses old composer (it will be implemented next, in another huge section - message display);
  4. Drafts being saved in single copy only, and proper draft discard;


That is it. Looking forward for your comments. :heart:

**Comment by JimTR** [comment]
Having used the theme for a while the only feed back is in some folders I get this
![email](https://user-images.githubusercontent.com/4038510/66470685-fdb9be00-ea81-11e9-9805-90bab5b83daf.png)
as you can see there are some ghost messages

**Comment by rostovtsev** [comment]
Are you using IMAP or Mailbox? You should see the same result in other mail clients and with older themes as well.

I guess, it's the stuck cache (broken indexes). In case you're using Dovecot, try:

~~~
doveadm force-resync -u <username> <mailbox>
~~~

https://wiki2.dovecot.org/Tools/Doveadm

**Comment by JimTR** [comment]
Tried that no difference but Thunderbird does not find the ghost entries
usermin read
![mail2](https://user-images.githubusercontent.com/4038510/66585838-7b172880-eb7f-11e9-9f28-aa6dc0b39ef4.png)
Thunderbird Read 
![thunder mail](https://user-images.githubusercontent.com/4038510/66585916-a39f2280-eb7f-11e9-8959-5fd0936ad45b.png)

**Comment by rostovtsev** [comment]
Do you get the same result in Gray Theme?

**Comment by JimTR** [comment]
Yes it does happen on all themes

**Comment by rostovtsev** [comment]
Well, try to stop Dovecot, delete indexes under:

```
~/Maildir/
```

..and restart Dovecot. It will work in case your _Mail storage format for Inbox_ option in _Webmin/Usermin Configuration/Usermin Module Configuration/Inbox format and location_ set to _Remote IMAP server_.

**Comment by gnadelwartz** [comment]
I updated to latest usermin and authentic to see the newest mail ux.
are the many Buttons as they are intended?
![Screenshot_20191220_104900](https://user-images.githubusercontent.com/4593242/71246123-63de7f00-2316-11ea-8bfb-9fb95ca42db8.jpg]

as I now see the button translation, I'll make some adjustments to be clearer and shorter if possible.

**Comment by rostovtsev** [comment]
Kay, hi.

Thanks for your feedback.

About buttons in mail preview - it's still an old, not-reworked section.

New mail UI is being released in parts. First was mail preview, and just recently I added mail compose.

New mail composer includes scheduling emails right from the composer, the way it's done in GMail.

https://user-images.githubusercontent.com/4426533/64477911-89051280-d1a9-11e9-8e71-da763fc5366f.png

**Comment by gnadelwartz** [comment]
after update from repo and restart I get the new composer which was not the case before. thx!

**Comment by rostovtsev** [comment]
Tell me how the new composer feels like for you, if you have a chance to run tests.

If GPG is enabled in _Usermin Configuration_ you will also see GPG menu entry there.

**Comment by gnadelwartz** [comment]
when creating a new message and want to change priority I get this:
![Screenshot_20191221_183005](https://user-images.githubusercontent.com/4593242/71311604-5ca39800-2420-11ea-9cd0-74b286e66cc2.jpg)

may be missing strings?

**Comment by rostovtsev** [comment]
Is it when your run Usermin in German? What about default English, does it work for you?

**Comment by gnadelwartz** [comment]
in english the strings are shown correct.

I also found that the mail folders are not shown after login. I have to switch to dashboard and back to mail. then inbox, sent etc sre shown.

**Comment by gnadelwartz** [comment]
found also an issue with language handling with new nessage.

- when language setting from webmin and in usermin is the same all massages are displayed e.g. in german.
- when language is set from webmin -> usermin settings to English utf-8 (not based on browser) and language set to german in usermin you get mixed language.
![Screenshot_20191222_081722](https://user-images.githubusercontent.com/4593242/71318684-944c2780-2494-11ea-81ea-997272bc4152.jpg)

all regular usermin messages are german as expected, but messages in new Mail UX are mixed ...

**Comment by rostovtsev** [comment]
Kay, does the same happen if used with  Gray Theme?

**Comment by gnadelwartz** [comment]
I can test it, but I'm shure it does only affect new mail UX, as e.g. old 
Mail preview is not affected ...

**Comment by rostovtsev** [comment]
There were fix issues that I fixed:

 1. Incorrect UI language returned (date select didn't work in local lang)
 2. Added page reload with nice UI message, upon language change
 3. Fix priority missing labels in other languages https://github.com/authentic-theme/authentic-theme/commit/9c61790c9979ca99e74930dc8f66af65f7c84821

Thanks for reporting!

**Comment by superlgn** [comment]
In gnadelwartz screenshot of the new composer, there appears to be an inline address book icon on the right side of the recipient field. Is that still present in 19.45? I'm not seeing it or any way to select someone from the address book when composing a new message.

**Comment by rostovtsev** [comment]
This hasn't been implemented yet.

**Comment by TwoLifestyles** [comment]
It's already 2023, we still get "forward functionality no implemented yet" error when trying to forward an email.
Will this be fixed soon?

**Comment by iliajie** [comment]
> It's already 2023, we still get "forward functionality no implemented yet" error when trying to forward an email.

When do you see this message exactly? How can I reproduce it on my side?

**Comment by TwoLifestyles** [comment]
Simply log into the read mail module of Usermin, check a message, and click the forward button at the top.
Screenshots show what happens.
<img width="897" alt="Screenshot 2023-09-02 at 8 04 32 AM" src="https://github.com/webmin/authentic-theme/assets/119398373/9c3fb2ab-bd9e-4289-9100-0e5963b77fd1">
<img width="897" alt="Screenshot 2023-09-02 at 8 04 40 AM" src="https://github.com/webmin/authentic-theme/assets/119398373/12a9048e-16b1-4918-9b8c-dff901f2a47a">
<img width="693" alt="Screenshot 2023-09-02 at 8 05 03 AM" src="https://github.com/webmin/authentic-theme/assets/119398373/03f7b920-5f3c-4162-a22e-3bf641ab5ee6">

**Comment by ksihota** [comment]
Forwarding displays properly when you are in the message but not when you do it this way

![image](https://github.com/webmin/authentic-theme/assets/29150649/9680973c-699a-4281-b8db-e1917976030c)

**Comment by iliajie** [comment]
Alright, I see what you mean. It will be fixed in the next Authentic Theme 21.06.

Thanks for the heads up, as I also discovered and fixed a few other event-related bugs!

**Comment by TwoLifestyles** [comment]
Thank you!
One final question, will you be adding the ability to add a picture or attachment within the signature section?
We have some good signatures created in Canva, but we can;t use them because there is no where to attach a file to the signature, or add one with the text field.
Maybe a proper WYSIWYG such as TinyMCE or other similar editor with the ability to attach images would help.

**Comment by iliajie** [comment]
> We have some good signatures created in Canva, but we can't use them because there is no where to attach a file to the signature, or add one with the text field.

Sounds like a good improvement. Although, I'm not sure if I have time to work on this particular feature any time soon. There is a lot to do.

But I will add this to my todo.

**Comment by TwoLifestyles** [comment]
Thank You!

**Comment by shoulders** [comment]
I know this is an old thread, the image in the first thread looks great, modern and clean.

**Comment by TwoLifestyles** [comment]
BUMP
We are still waiting for this feature three years later.
Not having it is the one thing preventing us from switching from roundcube. 

"
Will you be adding the ability to add a picture or attachment within the signature section?
We have some good signatures created in Canva, but we can't use them because there is no where to attach a file to the signature, or add one with the text field.
Maybe a proper text editor such as WYSIWYG, or other similar editor with the ability to attach images would help.
"

**Comment by iliaross** [comment]
> BUMP

Thanks!

> Will you be adding the ability to add a picture or attachment within the signature section?

Do you mean a signature that's automatically added to every new message or reply?

**Comment by TwoLifestyles** [comment]
Yes. When you click the edit signature button. Its just a text field. It has no markup, or places to add an image. 
Nowadays most company signatures are fancy images that get added to emails, and not having advanced edit capabilities here prevents us from using usermin. 

At the very least, add TinyMCE here so we can do HTML signatures easily and add images to it from there. 
Thank You :)

<img width="2375" height="646" alt="Image" src="https://github.com/user-attachments/assets/047e278f-9d5a-4cd2-ae18-409bfe328288" />

**Comment by shoulders** [comment]
@TwoLifestyles I think you should raise this as a separate issue. This issue is about a complete design rework rather than and individual component. But an upgrade to the signature thing might be useful.

@iliaross Maybe this issue should be closed as it was about implementing the theme in the first port (which I like) however I am presuming you are not going down this route.

**Comment by iliaross** [comment]
> Nowadays most company signatures are fancy images that get added to emails, and not having advanced edit capabilities here prevents us from using usermin.

It's been added in the latest development build:

<img width="1239" height="986" alt="Image" src="https://github.com/user-attachments/assets/1c525539-41e8-4ee4-a666-11279499ab12" />

Please give it a try:
https://download.webmin.dev/usermin-latest.deb
https://download.webmin.dev/usermin-latest.rpm
